File: app/src/main/java/com/soft/credit911/Dashboard/fragment/DashboardFragment.java
package com.soft.credit911.Dashboard.fragment;


import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;


import com.soft.credit911.Dashboard.Model.DashboardResponse;
import com.soft.credit911.Dashboard.mvp.DashboardPresenter;
import com.soft.credit911.Dashboard.mvp.DashboardView;
import com.soft.credit911.Landing.LandingActivity;
import com.soft.credit911.SecurityQuestions.Activity.SecurityQuestionsActivity;
import com.soft.credit911.SecurityQuestions.Model.SecurityResponse;
import com.soft.credit911.Utils.CommonUtils;
import com.soft.credit911.databinding.FragmentDashboardBinding;
import com.soft.credit911.databinding.ToolbarBinding;


public class DashboardFragment extends Fragment implements DashboardView {
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    private FragmentDashboardBinding fragmentBinding;
    private ToolbarBinding toolbarBinding;
    DashboardPresenter dashboardPresenter;

    @SuppressLint("SetTextI18n")
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        fragmentBinding = FragmentDashboardBinding.inflate(inflater, container, false);
        toolbarBinding = fragmentBinding.toolbarLayout;
        toolbarBinding.toolbarTitle.setText("Dashboard");
        toolbarBinding.navigationIcon.setVisibility(View.GONE);
        dashboardPresenter = new DashboardPresenter(getContext(), this);
        fragmentBinding.tvCreditRepairStatus.setOnClickListener(v -> {
            Intent intent = new Intent(getContext(), SecurityQuestionsActivity.class);
            getContext().startActivity(intent);
        });
        fragmentBinding.sesameView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                fragmentBinding.sesameView.setSesameValues(1000);
            }
        });

        fragmentBinding.scroolableContent.setVisibility(View.VISIBLE);
        fragmentBinding.txtErrorText.setVisibility(View.GONE);
        if (getArguments() != null) {
            if(getArguments().containsKey("isError") && getArguments().getInt("isError")==1){
                fragmentBinding.scroolableContent.setVisibility(View.GONE);
                fragmentBinding.txtErrorText.setVisibility(View.VISIBLE);
            }else{
                dashboardPresenter.creditReport();
            }
        }else{
            dashboardPresenter.creditReport();
        }

        return fragmentBinding.getRoot();
    }

    @Override
    public void DashboardResponse(DashboardResponse dashboardResponse) {
       CommonUtils.showdialog(dashboardResponse.getMessage(), getContext(), false);
        String reportDate = dashboardResponse.getData().getCreditReport().getReportDate();
        String nextDate = dashboardResponse.getData().getCreditReport().getNextDate();
        fragmentBinding.tvReportDate.setText(reportDate);
        fragmentBinding.tvNextDate.setText(nextDate);

    }

    @Override
    public void SecurityResponse(SecurityResponse securityResponse) {
        if(securityResponse.getCode().equals("100")){
            String token = securityResponse.getData().getAuthToken();
            String jsResponse = securityResponse.getData().getQuestions();
            Intent intent = new Intent(getContext(), SecurityQuestionsActivity.class);
            intent.putExtra("responseObj",jsResponse);
            intent.putExtra("token",token);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        }
    }

  /*  private final int[] mColors = new int[]{
            0xFFFF80AB,
            0xFFFF4081,
            0xFFFF5177,
            0xFFFF7997
    };


   public void startColorChangeAnim() {

        ObjectAnimator animator = ObjectAnimator.ofInt(fragmentBinding.layout, "backgroundColor", mColors);
        animator.setDuration(3000);
       animator.setEvaluator(new ArgbEvaluator());
        animator.start();
    }*/
}

File: app/src/main/java/com/soft/credit911/ui/dashboard/Dashboard/DashboardFragment.kt
package com.soft.credit911.ui.dashboard.Dashboard

import android.content.Intent
import android.graphics.Color
import android.util.Log
import android.view.View
import androidx.lifecycle.Observer
import com.github.mikephil.charting.components.*
import com.ing.quiz.ui.base_classes.BaseFragment
import com.soft.credit911.ChartUtils
import com.soft.credit911.R
import com.soft.credit911.Utils.CommonUtils.Companion.showdialog
import com.soft.credit911.datamodel.DashboardResponse
import com.soft.credit911.ui.SecurityQuestions.SecurityQuestionsActivity
import com.soft.credit911.ui.dashboard.ActivityCreditHistory
import com.soft.credit911.ui.dashboard.ActivityCreditReport
import com.soft.credit911.ui.dashboard.LandingActivity
import com.soft.credit911.ui.documnet.DocumentActivity
import kotlinx.android.synthetic.main.fragment_dashboard.*
import kotlinx.android.synthetic.main.toolbar.*
import java.util.*
import kotlin.collections.ArrayList


class DashboardFragment : BaseFragment() {

    var viewModel: DashBoardViewModel? = null
    var historyData:ArrayList<DashboardResponse.CreditReportHistoryItem>?=null
    var creditData:DashboardResponse.CreditReport?=null

    override fun getLayoutID(): Int {
        return R.layout.fragment_dashboard
    }

    override fun onViewCreated() {

    }

    override fun onResume() {
        super.onResume()
        setInformation()
    }

    fun setInformation() {
        toolbarTitle.text = "Dashboard"
        navigationIcon.visibility = View.GONE
        viewModel = DashBoardViewModel()
        attachObserver()

        tv_credit_repair_status.setOnClickListener { v: View? ->
            (activity as LandingActivity).selectCaseScreen()
        }

        tv_credit_history.setOnClickListener {
            if(historyData!=null) {
                val intent = Intent(activity, ActivityCreditHistory::class.java)
                intent.putExtra("history", historyData)
                startActivity(intent)
            }
        }

        tv_credit_report.setOnClickListener {
            if(creditData!=null) {
                val intent = Intent(activity, ActivityCreditReport::class.java)
                intent.putExtra("creditData", creditData)
                startActivity(intent)
            }
        }

        scroolableContent.visibility = View.VISIBLE
        txtErrorText.visibility = View.GONE
        if (arguments != null) {
            if (arguments?.containsKey("isError")==true && arguments?.getInt("isError") == 1) {
                scroolableContent.visibility = View.GONE
                txtErrorText.visibility = View.VISIBLE
            } else {
                viewModel?.getCreditInfo()
            }
        } else {
            viewModel?.getCreditInfo()
        }

        CardToDoc.setOnClickListener {
            val intent = Intent(activity, DocumentActivity::class.java)
            activity?.startActivity(intent)
        }
    }

    fun attachObserver(){

        viewModel?.isLoading?.observe(viewLifecycleOwner, Observer {
            if(it){
                showProgress()
            }else{
                hideProgress()
            }

        })
        viewModel?.responseAppHomedata?.observe(viewLifecycleOwner, Observer {dashboardResponse->
            if(dashboardResponse.message!=null && dashboardResponse?.message?.length?:0>0){
                context?.let { showdialog(dashboardResponse.message, it, false) }
            }

            val reportDate = dashboardResponse.data!!.creditReport!!.reportDate
            val nextDate = dashboardResponse.data!!.creditReport!!.nextDate
            historyData=dashboardResponse.data?.creditReportHistory
            creditData=dashboardResponse?.data?.creditReport
            tv_report_date.text = reportDate
            tv_next_date.text = nextDate
            setChartData()
            scroolableContent.visibility=View.VISIBLE
            score.text=dashboardResponse?.data?.creditReport?.score?:"0"
            val sscore=dashboardResponse.data?.creditReport?.score?.toInt()?:0
            if(sscore<400){
                scoreAbout.text="POOR"
            }
            else if(sscore<700){
                scoreAbout.text="NEEDS WORK"
            }else{
                scoreAbout.text="GOOD"
            }
            ChartUtils().setChartData(chart,dashboardResponse?.data?.creditReportHistory)
            if(sscore<=300){
                progress.setProgress(0.01f)
            }else if (sscore<=500){
                progress.setProgress((sscore-300)/1000.0f)
            }
            else if (sscore<=575){
                progress.setProgress(.2f+(sscore-500f)*0.0027f)
            }
            else if (sscore<=650){
                progress.setProgress(.4f+(sscore-575f)*0.0033f)
            }
            else if (sscore<=750){
                progress.setProgress(.65f+(sscore-650f)*0.0015f)
            }
            else if (sscore<=900){
                progress.setProgress(.8f+(sscore-750f)*0.0015f)
            }
            CardToDoc.visibility=View.GONE
            if(dashboardResponse?.data?.document_alert!=null
                && dashboardResponse?.data?.document_alert?.alert_status.equals("y")){
                CardToDoc.visibility=View.VISIBLE
                textMessage.text= dashboardResponse?.data?.document_alert?.message
                CardToDoc.setCardBackgroundColor(Color.parseColor(dashboardResponse?.data?.document_alert?.color_code))
            }

        })

        viewModel?.responseSecurity?.observe(viewLifecycleOwner, Observer {securityResponse->
            if (securityResponse.code == "100") {
                val token = securityResponse.data!!.authToken
                val jsResponse = securityResponse.data!!.questions
                val intent = Intent(context, SecurityQuestionsActivity::class.java)
                intent.putExtra("responseObj", jsResponse)
                intent.putExtra("token", token)
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
                startActivity(intent)
            }
        })
    }


    fun setChartData(){

    }

}
